### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by andriy-shymkiv**
[DEMO LINK](https://andriy-shymkiv.github.io/react_todo-app-add-and-delete/)

### CODE REVIEW

**Comment by Andriy-Shymkiv** [comment]
> [DEMO LINK](https://andriy-shymkiv.github.io/react_todo-app-add-and-delete/)
> 
> Almost done, i just can't fix a bug with loader while adding todo

**Comment by Andriy-Shymkiv** [comment]
> looks fine. I gonna steal a couple of your ideas i you don't mind

File: src/App.tsx [hunk 1]
```diff
@@ -12,212 +33,113 @@ export const App: React.FC = () => {
...
+
+    const getTodosFromServer = async () => {
+      const todosFromServer = user && await getTodos(user.id);
+
+      try {
+        if (user) {
+          setTodos(await getTodos(user.id));
+        }
+
+        // condition to show footer
```
Comment (mykola-koval) [inline src/App.tsx:45]: remove comments

File: src/App.tsx [hunk 2]
```diff
@@ -12,212 +33,113 @@ export const App: React.FC = () => {
...
+        if (todosFromServer && todosFromServer.length > 0) {
+          setShowFooter(true);
+        }
+      } catch {
+        setError(ErrorType.NoTodos);
+
+        setTimeout(() => {
+          setError(ErrorType.None);
+        }, 3000);
```
Comment (mykola-koval) [inline src/App.tsx:54]: looks like this code can be reused in several places

File: src/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  onTodosChange,
+  todos,
+}) => {
+  const [activeTodos, setActiveTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<SortBy>();
+  const [hasCompletedTodos, setHasCompletedTodos] = useState<boolean>();
+
+  const user = useContext(AuthContext);
+
+  // to check if there are some completed todos
```
Comment (mykola-koval) [inline src/components/Footer/Footer.tsx:28]:
```suggestion
```

File: src/components/Footer/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+      return todosFromServer
+        && todosFromServer.filter(todo => todo.completed).length > 0
+        ? setHasCompletedTodos(true)
+        : setHasCompletedTodos(false);
+    };
+
+    completedTodos();
+  }, []);
+
+  // to find active todos (bottom left corner)
```
Comment (mykola-koval) [inline src/components/Footer/Footer.tsx:42]:
```suggestion

```

File: src/components/Footer/Footer.tsx [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+          href="#/"
+          className={cn(
+            'filter__link',
+            {
+              selected: filterBy === All,
+            },
+          )}
+          onClick={() => {
+            handleFilter(All);
+            setFilterBy(All);
```
Comment (mykola-koval) [inline src/components/Footer/Footer.tsx:112]: why you need both?

Comment (mykola-koval) [reply src/components/Footer/Footer.tsx:112]: let's combine them

**Review by mykola-koval — APPROVED** [review]

**Pull request closed without merging**